src/lava/lib/optimization/solvers/generic/solver.py

```diff
@@ -82,6 +84,35 @@ def __init__(self,
         self._process_builder = SolverProcessBuilder()
         self.solver_process = None
         self.solver_model = None
+        shape = (problem.num_variables,)
+        self._hyperparameters = dict(step_size=10,
+                                     steps_to_fire=10,
+                                     noise_amplitude=1,
+                                     init_value=np.zeros(shape),
+                                     init_state=np.zeros(shape))
+        self._report = dict(solved=None,
+                            best_state=None,
+                            cost=None,
+                            target_cost=None,
+                            steps_to_solution=None,
+                            time_to_solution=None,
+                            power_to_solution=None)
+        self._benchmarker = SolverBenchmarker()
+
+    @property
+    def measured_time(self):
+        sts = self.last_run_report['steps_to_solution']
+        print(f"{self._benchmarker.measured_time.sum()=}")
+        print(f"{repr(self._benchmarker.measured_time)=}")
+        import matplotlib.pyplot as plt

```

Comment:
Plotting should only be done if a flag has been set. As default, I would suggest to deactivate it. No unnecessary output as default, in particular not graphics output.

Reply:
Agree!